basicapp/spec.go

```diff
@@ -0,0 +1,81 @@
+package basicapp
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"k8s.io/client-go/kubernetes"
+)
+
+type Clients interface {
+	// K8sClient returns a properly configured control plane client for the
+	// Kubernetes API.
+	K8sClient() kubernetes.Interface
+}
+
+// Chart is the chart to test.
+type Chart struct {
+	Name        string
+	URL         string
+	ChartValues string
+	Namespace   string
+}
+
+func (cc Chart) Validate() error {
+	if cc.URL == "" {
+		return microerror.Maskf(invalidConfigError, "%T.URL must not be empty", cc)
+	}
+	if cc.Name == "" {
+		return microerror.Maskf(invalidConfigError, "%T.Name must not be empty", cc)
+	}
+	if cc.Namespace == "" {
+		return microerror.Maskf(invalidConfigError, "%T.Namespace must not be empty", cc)
+	}
+
+	return nil
+}
+
+// ChartResources are the key resources deployed by the chart.
+type ChartResources struct {
+	DaemonSets  []DaemonSet
+	Deployments []Deployment
+}
+
+func (cr ChartResources) Validate() error {
+	if len(cr.DaemonSets) == 0 && len(cr.Deployments) == 0 {
+		return microerror.Maskf(invalidConfigError, "at least one daemonset or deployment must be specified")
+	}
+
+	return nil
+}
+
+// DaemonSet is a daemonset to be tested.
+type DaemonSet struct {
+	Name        string
+	Namespace   string
+	Labels      map[string]string
+	MatchLabels map[string]string
+	Replicas    int
```

Comment:
Same as above. I think we can drop the replicas and just use the value we specify in the chart.